arch/arm/boot/dts/imx6ul-ts7100.dtsi

```diff
@@ -0,0 +1,543 @@
+				MX6UL_PAD_CSI_DATA05__EIM_AD05		0x1b029
+				MX6UL_PAD_CSI_DATA04__EIM_AD04		0x1b029
+				MX6UL_PAD_CSI_DATA03__EIM_AD03		0x1b029
+				MX6UL_PAD_CSI_DATA02__EIM_AD02		0x1b029
+				MX6UL_PAD_CSI_DATA01__EIM_AD01		0x1b029
+				MX6UL_PAD_CSI_DATA00__EIM_AD00		0x1b029
+				/* FPGA IRQ */
+				MX6UL_PAD_SNVS_TAMPER1__GPIO5_IO01	0x1b029
+			>;
+		};
+
+		pinctrl_wdog: wdoggrp {
+			fsl,pins = <
+				MX6UL_PAD_LCD_RESET__WDOG1_WDOG_ANY    0x30b0
+			>;
+		};
+	};
+};
+
+&lcdif {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_lcd_ctrl>;
+	lcd-supply = <&reg_lcdif_enable>;
+	status = "okay";
+
+	port {
+		display_out: endpoint {
+			remote-endpoint = <&panel_in>;
+		};
+	};
+};
+
+&snvs_rtc {
+	status = "disabled";
+};
+
+&uart1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_uart1>;
+	status = "okay";
+};
+
+&uart7 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_uart7>;
+	status = "okay";
+};
+
+&usdhc1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_usdhc1>;
+	no-1-8-v;
+	disable-wp;
+	broken-cd = <1>;
+	bus-width = <4>;
+	status = "okay";
+};
+
+&wdog1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_wdog>;
+	status = "okay";
+	fsl,ext-reset-output;
+};
+
+&weim {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_weim_fpga>;
+	clocks = <&clks IMX6UL_CLK_EIM>, <&clks IMX6UL_CLK_EIM_SLOW_SEL>;
+	ranges = <0 0 0x50000000 0x08000000>;
+	status = "okay";
+
+	#address-cells = <2>;
+	#size-cells = <1>;
+
+	fpga: fpga@50000000 {
+		compatible = "simple-bus";
+		reg = <0 0x50000000 0x00010000>;
+		#address-cells = <1>;
+		#size-cells = <1>;
+		ranges = <0 0 0 0x10000>;
+
+		fsl,weim-cs-timing = <
+			0x0161030F	// EIM_CSnGCR1 @ 0x021b8000
+			0x00000000	// EIM_CSnGCR2 @ 0x021b8004
+			0x03000000	// EIM_CSnRCR1 @ 0x021b8008
+			0x00000000	// EIM_CSnRCR2 @ 0x021b800c
+			0x01000000	// EIM_CSnWCR1 @ 0x021b8010
+			0		// EIM_CSnWCR2 @ 0x021b8014
+		>;
+
+		clocks {
+			#address-cells = <1>;
+			#size-cells = <1>;
+			ranges;
+
+			fpga_clk_weim_bclk: fpga_clk_weim_bclk {
+				#clock-cells = <0>;
+				compatible = "fixed-clock";
+				clock-frequency = <49500000>;
+			};
+		};
+
+		syscon: syscon@4000 {
+			compatible = "simple-bus";
+			#address-cells = <0x2>;
+			#size-cells = <0x1>;
+			reg = <0x4000 0x58>;
+			ranges = <0 0 0x4000 0x58>;
+
+			gpio6: fpga_gpio6@10 {
+				compatible = "technologic,ts71xxweim-gpio";
+				reg = <0 0x10 0x08>;
+				gpio-controller;
+				#gpio-cells = <2>;
+				interrupt-controller;
+				#interrupt-cells = <1>;
+			};
+
+			gpio7: fpga_gpio7@40 {
+				compatible = "technologic,ts71xxweim-gpio";
+				reg = <0 0x40 0x08>;
+				gpio-controller;
+				#gpio-cells = <2>;
+				interrupt-controller;
+				#interrupt-cells = <1>;
+			};
+
+			gpio8: fpga_gpio8@50 {
+				compatible = "technologic,ts71xxweim-gpio";
+				reg = <0 0x50 0x08>;
+				gpio-controller;
+				#gpio-cells = <2>;
+				interrupt-controller;
+				#interrupt-cells = <1>;
+			};
+
+			fpga_intc: fpga_intc@0 {
```

Comment:
I _think_ this should be `fpga_intc: interrupt-controller@4024 {`

The address after the @ should be relative to the device, e.g. FPGA.